sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,417 @@
+            "default" if str(i[0]).lower() == "column_default" else str(i[0]).lower() for i in cursor.description
+        ]
+        rows = [dict(zip(columns, [str(item) for item in row])) for row in data]
+        for row in rows:
+            table_id = name_to_id.get(str(row.get("table_name")))
+            if table_id is not None:
+                row.pop("table_name", None)
+                if "nullable" in row:
+                    if row["nullable"].lower() == "no" or row["nullable"].lower() == "false":
+                        row["nullable"] = False
+                    else:
+                        row["nullable"] = True
+                if table_id in id_to_table_data:
+                    id_to_table_data.get(table_id)["columns"] = id_to_table_data.get(table_id).get("columns", []) + [
+                        row
+                    ]
+                else:
+                    self._log.error("Columns found for an unkown table with the object_id: %s", table_id)
+            else:
+                self._log.error("Couldn't find id of a table: %s", table_id)
+        return len(data)
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_partitions_data(self, table_ids, id_to_table_data, cursor):
+        rows = execute_query_output_result_as_dicts(PARTITIONS_QUERY.format(table_ids), cursor)
+        for row in rows:
+            id = row.pop("id", None)
+            if id is not None:
+                id_str = str(id)
+                if id_str in id_to_table_data:
+                    id_to_table_data[id_str]["partitions"] = row
+                else:
+                    self._log.error("Partition found for an unkown table with the object_id: %s", id_str)
+            else:
+                self._log.error("Return rows of [%s] query should have id column", PARTITIONS_QUERY)
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_index_data(self, table_ids, id_to_table_data, cursor):
+        rows = execute_query_output_result_as_dicts(INDEX_QUERY.format(table_ids), cursor)
+        for row in rows:
+            id = row.pop("id", None)
+            if id is not None:
+                id_str = str(id)
+                if id_str in id_to_table_data:
+                    id_to_table_data[id_str].setdefault("indexes", [])
+                    id_to_table_data[id_str]["indexes"].append(row)
+                else:
+                    self._log.error("Index found for an unkown table with the object_id: %s", id_str)
+            else:
+                self._log.error("Return rows of [%s] query should have id column", INDEX_QUERY)
+
+    @tracked_method(agent_check_getter=agent_check_getter, track_result_length=True)
+    def _populate_with_foreign_keys_data(self, table_ids, id_to_table_data, cursor):
+        rows = execute_query_output_result_as_dicts(FOREIGN_KEY_QUERY.format(table_ids), cursor)
+        for row in rows:
+            id = row.pop("id", None)
```

Comment:
these vars could have better naming to make this code more readable. I assume the `id` here is the table id? 

Also is `id_to_table_data` a  map of table id to data? I would probably name it as such in that case

Reply:
id - yes it's a what they call object_id in sqlserver so a table_id. Ok, will change.